docs/csharp/programming-guide/inside-a-program/coding-conventions.md

```diff
@@ -19,15 +19,16 @@ Coding conventions serve the following purposes:
 > - They facilitate copying, changing, and maintaining the code.
 > - They demonstrate C# best practices.
 
-The guidelines in this article are used by Microsoft to develop samples and documentation.
+> [!IMPORTANT]
+> The guidelines in this article are used by Microsoft to develop samples and documentation. They were adopted from the [.NET Runtime, C# Coding Style](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md) guidelines. You can use them, or adapt them to your needs. The primary objectives are consistency and readability within your project, team, organization, or company source code.
```

Comment:
>The guidelines in this article are used by Microsoft to develop samples and documentation

Hmm,... even the very basic example code in C# reference docs does not follow the newly added convention `s_`. It's not so popular outside runtime repo.
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/static#example---static-initialization

I think it's too late to add these new guidelines after 20 years have passed since C# was born.

Reply:
Consistency in official docs is so important. It needs a lot of work to correct these examples. Is it really worth to change at this time?

Reply:
Hi @SIkebe,

It's never too late to update guidelines. It is natural for them to morph over time - especially with a language that continues to evolve. Following the .NET runtime as a basis is a solid foundation, since it's open source and in the view of the public. Even within the runtime code base, you can find violations and inconsistencies - that's okay. There are bound to be inconsistencies, but those too (when and where applicable) could be addressed. Guidelines are a set of suggestions, not something that is mandated. You're free to ignore them.

Reply:
>It is natural for them to morph over time

Yeah, I agree if there is a decent need to do so, or strong feedback from users. Could you point to the discussion about this newly added convention? Users want to know "why" this happened at this time.

And how about @nanasi880 's point?
https://github.com/dotnet/docs/pull/24054#issuecomment-841094612

Reply:
> I think it's too late to add these new guidelines after 20 years have passed since C# was born.
I agree with this.

It's not that I don't like the change, but the reason for the change is unclear, and as I pointed out, this is a change that Microsoft has explicitly denied.

It seems that static variables have an 's_' prefix and thread-static variables have 't_', what about AsyncLocal for example, and ThreadLocal?

Also, what if there are more of these in the future?
What if fiber is born? Is 'f_' the preposition that fiber local storage has?

We believe that history has proven that distinguishing these things by prepositions is a bad idea.
This is exactly what we call Hungarian notation, and we don't think it's an idea we should adopt, at least not actively now.